File: Servidor/SWServidorLYJStore/SWServidorLYJStore/Project1/src/main/java/estructural/Producto.java
package estructural;

import java.io.Serializable;

import java.util.ArrayList;
import java.util.List;

public class Producto implements Serializable {
    @SuppressWarnings("compatibility:7218494295295732225")
    private static final long serialVersionUID = 1L;

    private Integer idProducto;
    private String nombre;
    private Double precioVenta;
    private Long upc;
    private List<Orden> ordenes;

    public Producto() {
        ordenes = new ArrayList<>();
    }


    public Producto(Integer idProducto) {
        this.idProducto = idProducto;
    }

    public Producto(Integer idProducto, String nombre, Double precioVenta, Long upc) {
        this.idProducto = idProducto;
        this.nombre = nombre;
        this.precioVenta = precioVenta;
        this.upc = upc;
    }


    public void setIdProducto(Integer idProducto) {
        this.idProducto = idProducto;
    }

    public Integer getIdProducto() {
        return idProducto;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getNombre() {
        return nombre;
    }

    public void setPrecioVenta(Double precioVenta) {
        this.precioVenta = precioVenta;
    }

    public Double getPrecioVenta() {
        return precioVenta;
    }

    public void setUpc(Long upc) {
        this.upc = upc;
    }

    public Long getUpc() {
        return upc;
    }

    public void setOrdenes(List<Orden> ordenes) {
        this.ordenes = ordenes;
    }

    public List<Orden> getOrdenes() {
        return ordenes;
    }
}

File: GUI PHP/testCleint/testCleint/Producto/producto.php
<?php

class producto {
    
    private $idProducto;
    private $nombre;
    private $precioVenta;
    private $upc;
    
        
    public function __construct($pID, $pNombre, $pPrecioVenta, $pUpc)
    {
        $this->idProducto = $pID;
        $this->nombre = $pNombre;
        $this->precioVenta = $pPrecioVenta;
        $this->upc = $pUpc;
        
    }
    
    public function getIdProducto()
    {
        return $this->idProducto;
       
    }
    
     public function setIdProducto(int $pID)
    {
        $this->idProducto = $pID;
    }
    
    public function getNombre()
    {
       return $this->nombre;
    }
    
     public function setNombre(string $pNombre)
    {
        $this->nombre = $pNombre;
    }
    
    public function getPrecioVenta()
    {
        return $this->precioVenta;
    }
    
     public function setPrecioVenta(int $precioVenta)
    {
        $this->precioVenta = $precioVenta;
    }
    
    public function getUpc()
    {
       return $this->upc;
    }
    
     public function setUpc(int $pUpc)
    {
        $this->upc = $pUpc;
    }
        
    
    
    
}
